src/types.rs

```diff
@@ -4,6 +4,7 @@ use core::fmt;
 pub type c_int = i32;
 pub type c_uchar = u8;
 pub type c_uint = u32;
+pub type c_char = i8;
```

Comment:
Whether `char` in C is signed or unsigned is inplementation-defined. That's why the actual definition rust of `c_char` is much more complex:  https://doc.rust-lang.org/src/std/os/raw/mod.rs.html#12
(Sad but that's how it is.)

Reply:
oh. that's really sad. hmmm what do you suggest? should we just copy the `raw` module here? Use https://crates.io/crates/cty ? (although adding another crate just for this isn't great)

we can maybe use travis to make sure the types are aligned with what's in the std of newer rust versions, although that won't make sure that it works on all platforms.

Reply:
I think just copying the definition from `raw` is simple and good enough.

Reply:
Done.
btw should I add the license comment to the `types.rs` file?

Reply:
Argh, I forgot that our license is CC0... Hm, that actually means that relying on a different crate is the cleanest solution but that introduces a dependency... @apoelstra @TheBlueMatt opinions?

Reply:
Is there any difference between manually copying code and just using the std library?

Reply:
Yes, it's a crucial difference:
If we copy it to our code, then we *distribute* in our repo claiming CC0 (unless we add exceptions to our LICENSE file or similar, which we would need to do then and which is kind of ugly).
If we rely a library, we won't distribute the library code. (But instead, whoever builds the library will need to obtain that code.)

Reply:
hmm, code law isn't my stronger side

Reply:
Do we use `c_char`?

Reply:
We can't use `cty` because it doesn't support our compiler version BTW.

Reply:
yeah, in the callbacks

Reply:
I think this whole argument is philosophical. It doesn't matter what type we alias as as long as it has the right size. It's not like we're adding `c_char`s and might have UB or not depending on signedness.

Let's just go back to the "assume `i8`" code.

Reply:
I think that even if in C `c_char` is u8, and it's set to ^127 it still won't be UB, because in Rust overflowing signed integers is defined.

Reply:
Regardless let's add a comment (a doccomment if we can) telling users not to use this type in their own code and that it might not match the C FFI exactly.